source/numerics/data_postprocessor.cc

```diff
@@ -196,6 +196,36 @@ DataPostprocessorTensor<dim>::get_needed_update_flags() const
 
 
 
+namespace DataPostprocessors
+{
+  template <int dim>
+  BoundaryIds<dim>::BoundaryIds()
+    : DataPostprocessorScalar<dim>("boundary_id", update_quadrature_points)
+  {}
+
+
+  template <int dim>
+  void
+  BoundaryIds<dim>::evaluate_scalar_field(
+    const DataPostprocessorInputs::Scalar<dim> &inputs,
+    std::vector<Vector<double>> &               computed_quantities) const
+  {
+    AssertDimension(computed_quantities.size(), inputs.solution_values.size());
+
+    const typename DoFHandler<dim>::active_cell_iterator cell =
+      inputs.template get_cell<dim>();
+    const unsigned int face = inputs.get_face_number();
+
+    for (auto &output : computed_quantities)
+      {
+        AssertDimension(output.size(), 1);
+        output(0) = cell->face(face)->boundary_id();
```

Comment:
What happens if this is run with ``DataOutFaces(/*surface_only=*/ false)``? Does this call fail with an assert? If not, can we document what values will be set in the interior?

Reply:
Ah, good question. 

I could go either way. My inclination is that something that only lives on the surface should only be used on the surface, and throw an assertion otherwise. Would you agree? If so, I'll implement that.

Reply:
I would think it is okay to run with internal faces and output -1 in the interior (documenting that would be great).
I imagine a situation where someone has other data to visualize in the interior like manifold ids.